dll/win32/kernel32/client/thread.c

```diff
@@ -946,9 +951,14 @@ SetThreadUILanguage(IN LANGID LangId)
     }
 #endif
 
-    UNIMPLEMENTED;
+    PreviousLcid = Teb->CurrentLocale;
+    PreviousLangId = LANGIDFROMLCID(PreviousLcid);
+
+    NewLcid = MAKELCID(LangId, SORT_DEFAULT);
+
+    Teb->CurrentLocale = NewLcid;
 
-    return LANGIDFROMLCID(NtCurrentTeb()->CurrentLocale);
+    return PreviousLangId;
```

Comment:
I doubt it returns the previouslangid, unless maybe it failed the suggested SetThreadLocale call above.